File: LeetCode/Arrays and Strings/Contains Duplicate/JavaScript Solution/ContainsDuplicate.js
var containsDuplicate = function(nums) {
    const hash = {};
    for(let num of nums) {
        hash[num] = hash[num] + 1 || 1;
    }
    
    for(let num in hash) {
        if(hash[num] > 1) {
            return true;
        }
    }
    
    return false;
};

console.log(containsDuplicate([3,1]))

File: LeetCode/Arrays and Strings/Contains Duplicate/C++ Solution/contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int count = 0; unordered_map<int, int> hash;
        for(int i = 0; i < nums.size(); i++) {
            hash[nums[i]]++;
        }
        for(auto count: hash) {
            if(count.second > 1) {
                return true;
            }
        }
        
        return false;
    }
};